webapp/channels/src/components/advanced_text_editor/send_button/send_post_options/core_menu_options.tsx

```diff
@@ -23,6 +30,69 @@ function CoreMenuOptions({handleOnSelect, channelId}: Props) {
         isDM,
     } = useTimePostBoxIndicator(channelId);
 
+    const recentlyUsedCustomDate = useSelector((state: GlobalState) => getPreference(state, scheduledPosts.SCHEDULED_POSTS, scheduledPosts.RECENTLY_USED_CUSTOM_TIME));
+
+    interface RecentlyUsedCustomDate {
+        update_at?: number;
+        timestamp?: number;
+    }
+
+    let recentlyUsedCustomDateVal: RecentlyUsedCustomDate = {};
+
+    if (recentlyUsedCustomDate) {
+        try {
+            recentlyUsedCustomDateVal = JSON.parse(recentlyUsedCustomDate) as RecentlyUsedCustomDate;
+        } catch (e) {
+            recentlyUsedCustomDateVal = {};
+        }
+    }
+
+    function isTimestampWithinLast30Days(timestamp: number, timeZone = 'UTC') {
+        if (!timestamp || isNaN(timestamp)) {
+            return false;
+        }
+        const usedDate = DateTime.fromMillis(timestamp).setZone(timeZone);
+        const now = DateTime.now().setZone(timeZone);
+        const thirtyDaysAgo = now.minus({days: 30});
+
+        return usedDate >= thirtyDaysAgo && usedDate <= now;
+    }
+
+    let recentCustomTime = null;
+    const handleRecentlyUsedCustomTime = useCallback((e) => handleOnSelect(e, recentlyUsedCustomDateVal.timestamp!), [handleOnSelect, recentlyUsedCustomDateVal.timestamp]);
+    if (
+        recentlyUsedCustomDateVal &&
+        typeof recentlyUsedCustomDateVal.update_at === 'number' &&
+        typeof recentlyUsedCustomDateVal.timestamp === 'number' &&
+        isTimestampWithinLast30Days(recentlyUsedCustomDateVal.update_at, userCurrentTimezone)
+    ) {
+        const timestamp = (
+            <Timestamp
+                value={recentlyUsedCustomDateVal.timestamp}
+                timeZone={userCurrentTimezone}
+                useRelative={false}
+                useDate={{weekday: 'long'}}
+                useTime={{hour: 'numeric', minute: 'numeric'}}
+            />
+        );
```

Comment:
In the same spirit about "being as stable as possible", we are passing a new timestamp on every render to `recentCustomTime`. Again, probably not a big deal, but we can try to be more stable.

We can refactor the logic to do something like...

```
const timestamp = useMemo(() => {
    if (!(recentlyUsedCustomDateVal && ...) {
        return undefined;
    }
    return (<Timestamp .../>);
, [...]);

const recentCustomTime = timestamp ? (
    <><Menu.Separator .../><Menu.Item .../></>
): undefined;
```

Reply:
seems overcomplicated for the potential gain in terms of performance. For these is hard to tell what is more beneficial, if let React do the calculation or keep that value in memory and the cost of checking if any of the dependencies change to be recreated. I am inclined to not do what you suggest.